src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -97,6 +97,56 @@ void Pet::RemoveFromWorld()
     }
 }
 
+SpellCastResult Pet::TryLoadFromDB(Player* owner, bool current /*= false*/, PetType mandatoryPetType /*= MAX_PET_TYPE*/)
+{
+    PreparedStatement* stmt = CharacterDatabase.GetPreparedStatement(CHAR_SEL_CHAR_PET_BY_ENTRY_AND_SLOT);
+    stmt->setUInt32(0, owner->GetGUIDLow());
+    stmt->setUInt8(1, uint8(current ? PET_SAVE_AS_CURRENT : PET_SAVE_NOT_IN_SLOT));
+
+    PreparedQueryResult result = CharacterDatabase.AsyncQuery(stmt);
+    
+    if (!result)
+        return SPELL_FAILED_NO_PET;
+
+    Field* fields = result->Fetch();
+
+    uint32 petentry = fields[1].GetUInt32();
+    uint32 savedHealth = fields[10].GetUInt32();
+    uint32 summon_spell_id = fields[15].GetUInt32();
+    PetType petType = PetType(fields[16].GetUInt8());
```

Comment:
not important, but for simplicity you can just use `auto petType = PetType(fields[16].GetUInt8());`

it's a good practice to use `auto` instead of repeating the type name twice, especially when they get quite long

Reply:
done!!